pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -118,6 +128,32 @@ func (admitter *VMICreateAdmitter) Admit(ar *admissionv1.AdmissionReview) *admis
 	return &reviewResponse
 }
 
+func (admitter *VMICreateAdmitter) validateName(vmi *v1.VirtualMachineInstance) []metav1.StatusCause {
```

Comment:
Even locking the VM and the VMI webhook would not allow making this race-free. There could always just have been a VM created or a VMI created. After a webhook is done, there is still work going on, the object is not persisted, that is why not even a lock would help.